bin/compile

```diff
@@ -1,28 +1,28 @@
 #!/bin/bash
 
-mkdir -p $1 $2 $3
-BUILD_PATH=$(cd $1 && pwd)
-CACHE_PATH=$(cd $2 && pwd)
+mkdir -p "$1" "$2" "$3"
+BUILD_PATH=$(cd "$1" && pwd)
+CACHE_PATH=$(cd "$2" && pwd)
 BUILDPACK_PREFIX=nmap
 
 NMAP_VERSION="7.60"
-if [ -f $3/NMAP_VERSION ]; then
-  NMAP_VERSION="$(cat $3/NMAP_VERSION)"
+if [ -f "$3"/NMAP_VERSION ]; then
+  NMAP_VERSION="$(cat "$3"/NMAP_VERSION)"
 fi
 NMAP_NAME=nmap-$NMAP_VERSION
-echo $CACHE_PATH/$BUILDPACK_PREFIX/$NMAP_NAME/bin/nmap
+echo "$CACHE_PATH"/"$BUILDPACK_PREFIX"/"$NMAP_NAME"/bin/nmap
 
-if [ ! -f $CACHE_PATH/$BUILDPACK_PREFIX/$NMAP_NAME/bin/nmap ]; then
+if [ ! -f "$CACHE_PATH"/"$BUILDPACK_PREFIX"/"$NMAP_NAME"/bin/nmap ]; then
   echo "-----> Installing $NMAP_NAME"
-  curl -s -O https://nmap.org/dist/$NMAP_NAME.tar.bz2
-  bzip2 -cd $NMAP_NAME.tar.bz2 | tar xf -
-  cd $NMAP_NAME
-  ./configure --prefix=$CACHE_PATH/$BUILDPACK_PREFIX/$NMAP_NAME
+  curl -s -O https://nmap.org/dist/"$NMAP_NAME".tar.bz2
+  bzip2 -cd "$NMAP_NAME".tar.bz2 | tar xf -
+  cd "$NMAP_NAME"
+  ./configure --prefix="$CACHE_PATH"/"$BUILDPACK_PREFIX"/"$NMAP_NAME"
   make
   make install
 else
   echo "-----> Using cached nmap version"
 fi
 
-mkdir -p $BUILD_PATH/$BUILDPACK_PREFIX/bin
-cp -r $CACHE_PATH/$BUILDPACK_PREFIX/$NMAP_NAME/bin/ $BUILD_PATH/$BUILDPACK_PREFIX/bin
+mkdir -p "$BUILD_PATH"/"$BUILDPACK_PREFIX"/bin
+cp "$CACHE_PATH"/"$BUILDPACK_PREFIX"/"$NMAP_NAME"/bin/* "$BUILD_PATH"/"$BUILDPACK_PREFIX"/bin
```

Comment:
Have you tried `cp -R` or `cp -a` (The -a option is an improved recursive option, that preserve all file attributes, and also preserve symlinks.) I'm not sure that `cp` without any args will work. If you think it's ok then we can consider it approved :)